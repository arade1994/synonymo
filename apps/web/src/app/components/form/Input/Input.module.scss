.inputContainer {
  position: relative;

  & .input {
    width: 100%;
    padding: 1rem;
    border: 1px solid #dcdcdc;
    border-radius: 8px;
    font-size: 1rem;
    transition: border-color 0.3s, box-shadow 0.3s;

    &:focus {
      outline: none;
      border-color: #6c63ff;
      box-shadow: 0 0 0 3px rgba(108, 99, 255, 0.2);
    }

    &::placeholder {
      color: #a0a0a0;
    }
  }
}
