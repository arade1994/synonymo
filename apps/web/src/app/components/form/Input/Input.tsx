"use client";

import cx from "classnames";

import Errors from "@/app/components/form/Errors/Errors";

import styles from "./Input.module.scss";

interface Props {
  id: string;
  placeholder: string;
  type: string;
  value: string | number;
  errors: string[];
  containerClassName?: string;
  inputClassName?: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}

const Input: React.FC<React.PropsWithChildren<Props>> = ({
  id,
  placeholder,
  type,
  value,
  errors,
  containerClassName,
  inputClassName,
  onChange,
}) => {
  return (
    <div className={cx(styles.inputContainer, containerClassName)}>
      <input
        className={cx(styles.input, inputClassName)}
        id={id}
        placeholder={placeholder}
        type={type}
        value={value}
        onChange={onChange}
      />
      <Errors errors={errors} />
    </div>
  );
};

export default Input;
