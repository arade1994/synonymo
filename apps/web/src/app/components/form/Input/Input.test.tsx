import { describe, expect, test, vi } from "vitest";

import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import Input from "./Input";

const mockOnChange = vi.fn();

describe("Input", () => {
  test("It should match a snapshot when rendered by default", () => {
    const { baseElement } = render(
      <Input
        errors={[]}
        id="test"
        placeholder="test"
        type="text"
        value="test"
        onChange={mockOnChange}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });

  test("It should update the value when the input is changed", async () => {
    render(
      <Input
        errors={[]}
        id="test"
        placeholder="test"
        type="text"
        value="test"
        onChange={mockOnChange}
      />
    );

    const input = screen.getByTestId("test");
    expect(input).toBeDefined();
    await userEvent.type(input, "test2");
    expect(mockOnChange).toHaveBeenCalled();
  });
});
