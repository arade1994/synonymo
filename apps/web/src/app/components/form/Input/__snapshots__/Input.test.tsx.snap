// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > It should match a snapshot when rendered by default 1`] = `
<body>
  <div>
    <div
      class="_inputContainer_18ce5e"
    >
      <input
        class="_input_18ce5e"
        id="test"
        placeholder="test"
        type="text"
        value="test"
      />
      <div
        class="_error_b9420b"
      />
    </div>
  </div>
</body>
`;
